src/zino/snmp.py

```diff
@@ -26,26 +31,41 @@ def _get_engine():
     return _local.snmp_engine
 
 
+@dataclass
+class MibObject:
+    oid: str
+    value: Union[str, int]
+
+
 class SNMP:
     """Represents an SNMP management session for a single device"""
 
+    NON_REPEATERS = 0
+
     def __init__(self, device: PollDevice):
         self.device = device
 
-    async def get(self, *object):
-        """SNMP-GETs a single value"""
-        query = [ObjectType(ObjectIdentity(*object))]
+    async def get(self, *oid: str) -> Union[MibObject, None]:
```

Comment:
how bout this?
```suggestion
    async def get(self, *oid: str) -> Union[MibObject, None]:
        """SNMP-GETs the given `oid`
        arguments:
        *oid: series of numeric or symbolic values representing nodes in the MIB tree
        """
```

Reply:
Then I suggest the docstring follows up with some usage examples that include symbolic names as well as numeric OIDs.

Reply:
Figured out the values pysnmplib allows are pretty specific. Reference: https://github.com/pysnmp/pysnmp/blob/bc1fb3c39764f36c1b7c9551b52ef8246b9aea7c/pysnmp/smi/rfc1902.py#L35-L49

the `*oid` arg is put directly into an ObjectIdentity object, so whatever is supported by ObjectIdentity is supported by the `get` function

If we want to include all the options ObjectIdentify allows, then referencing the pysnmplib code is prob best instead of just copying it.

Or we just limit what we include.  get("SNMPv2-MIB", "sysUpTime", 0) and get("1.3.6.1.2.1.1.3.0") are probably the only formats we would want to use? 

Suggestions?

Reply:
merhaps something along these lines
```suggestion
    async def get(self, *oid: str) -> Union[MibObject, None]:
        """SNMP-GETs the given `oid`
        arguments:
        *oid: values for defining an OID. For detailed use see https://github.com/pysnmp/pysnmp/blob/bc1fb3c39764f36c1b7c9551b52ef8246b9aea7c/pysnmp/smi/rfc1902.py#L35-L49
        
       Example usage:
       get("SNMPv2-MIB", "sysUpTime", 0)
       get("1.3.6.1.2.1.1.3.0")
        """
```

Reply:
Yes, a reference and a limited number of examples (as we use in the codebase already) is perfect.

But also, we should be mindful about the formatting of docstrings, as we will likely produce documentation using Sphinx, as we do in other projects.  See https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html for how to format docstrings for arguments and return values (The bits about annotation of variable types is likely obsolete, since this has become part of the Python language by now).

Reply:
Having one docstring format to follow is very helpful. We should include this in the README
```suggestion
    async def get(self, *oid: str) -> Union[MibObject, None]:
        """SNMP-GETs the given `oid`
       Example usage:        
           get("SNMPv2-MIB", "sysUpTime", 0)
           get("1.3.6.1.2.1.1.3.0")
           
        :param *oid:  Values for defining an OID. For detailed use see https://github.com/pysnmp/pysnmp/blob/bc1fb3c39764f36c1b7c9551b52ef8246b9aea7c/pysnmp/smi/rfc1902.py#L35-L49
        :return: A MibObject matching *oid or None if nothing could be found
        
```

Reply:
Yup. But also, the correct variable reference is `oid`, not `*oid`.  The asterisk is part of the method signature and only tells Python to consume all remaining non-named arguments into `oid`.